src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -2242,7 +2238,7 @@ private static bool FilterApplyPrefixLookup(MemberInfo memberInfo, string name,
             if (!FilterApplyBase(type, bindingFlags, isPublic, type.IsNestedAssembly, isStatic: false, name, prefixLookup))
                 return false;
 
-            if (ns != null && ns != type.Namespace)
+            if (!Unsafe.IsNullRef(ref ns._reference) && !ns.SequenceEqual(type.Namespace))
```

Comment:
We generally don't distinguish between `ReadOnlySpan<char>.Empty`/`default` and other spans that are `IsEmpty`. I'm concerned this is too fragile.

Reply:
> We generally don't distinguish between `ReadOnlySpan<char>.Empty`/`default` and other spans that are `IsEmpty`. I'm concerned this is too fragile.

Thank you for clarification. I will add a boolean value to flag if there's a namespace, and check if the plain case (no NS) is further impacted. If the plain case is slower, I would like to close this PR.